client/src/test/java/com/paypal/selion/platform/grid/SessionSharingTestWithConfigurationMethods.java

```diff
@@ -0,0 +1,124 @@
+/*-------------------------------------------------------------------------------------------------------------------*\
+|  Copyright (C) 2015 PayPal                                                                                          |
+|                                                                                                                     |
+|  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.platform.grid;
+
+import static org.testng.Assert.assertTrue;
+
+import org.openqa.selenium.remote.SessionId;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterGroups;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeGroups;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import com.paypal.selion.TestServerUtils;
+import com.paypal.selion.annotations.WebTest;
+import com.paypal.selion.reports.runtime.SeLionReporter;
+
+@WebTest(browser = "chrome")
+@Test(singleThreaded = true)
+public class SessionSharingTestWithConfigurationMethods {
+    @BeforeSuite
+    public void beforeSuite() {
+        // not started yet
+        Assert.assertNull(Grid.getTestSession());
+    }
+
+    @BeforeTest
```

Comment:
Did you try multiple @BeforeTest, multiple etc?  The logic looks right in SeleniumGridListener but just wanted to double check.